ZeCamp/ZeCamp/Extensions/Sequence+groupBy.swift

```diff
@@ -0,0 +1,14 @@
+import Foundation
+
+public extension Sequence {
+    func group<U: Hashable>(by key: (Iterator.Element) -> U) -> [U:[Iterator.Element]] {
+        var categories: [U: [Iterator.Element]] = [:]
+        for element in self {
+            let key = key(element)
+            if categories[key]?.append(element) == nil {
+                categories[key] = [element]
+            }
+        }
+        return categories
```

Comment:
This doesn’t even exist in the beta 4 yet, but FYI, a new overload of `reduce` [is being added](https://github.com/apple/swift-evolution/blob/master/proposals/0171-reduce-with-inout.md).

With that, you can reduce  (pun intended 😄) the boilerplate into:
```swift
return reduce(into: [:]) { categories, element in
    let key = key(element)
    if categories[key]?.append(element) == nil {
        categories[key] = [element]
    }
}
```